mmcls/datasets/samplers/distributed_sampler.py

```diff
@@ -15,7 +15,8 @@ def __init__(self,
                  rank=None,
                  shuffle=True,
                  round_up=True,
-                 seed=0):
+                 seed=0,
+                 device='cuda'):
```

Comment:
Can we use `mmcls.utils.auto_select_device` here? Then we don't need to modify the `build_dataloader` function and don't need this argument.